src/node_api.cc

```diff
@@ -710,7 +724,11 @@ napi_status napi_define_class(napi_env env,
                                     v8impl::FunctionCallbackWrapper::Invoke,
                                     cbdata,
                                     v8::Signature::New(isolate, tpl));
-      t->SetClassName(property_name);
+
+      // Class name has to be an identifier, though property names can be any
+      // property key. Enforce that class name is a string here.
+      RETURN_STATUS_IF_FALSE(property_name->IsString(), napi_string_expected);
+      t->SetClassName(property_name.As<v8::String>());
```

Comment:
Sounds right. I was also confused about why we were calling ```SetClassName()``` here. Seems that is for constructors, not methods.